src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -892,6 +894,26 @@ public static object GetDataOfValue(Mirror.MirrorData value)
             return value.StringData;
         }
 
+        public static object GetDataOfCLRValue(object value)
+        {
+                // This is because stackvalues directly store value types
+                // while they store the string representation of reference types.
+                if (value is long || value is int || value is double || value is char || value is bool)
+                {
+                    return value;
+                }
+                else if (value is DesignScript.Builtin.Dictionary)
+            {
+                throw new NotImplementedException("get clr value of DSDictionary.");
+            }
+                //TODO expand.
+                else if(value is ICollection vc)
+            {
+                return vc.Cast<Object>().Select(x => GetDataOfCLRValue(x)).ToList();
```

Comment:
👍

Reply:
Is `ICollection` the only interface that allows iterations that might be found here ?
Or should we use IEnumerable ?

Reply:
I think right now, we always return an `IList` so thats an ICollection, and array also implement ICollection:
https://docs.microsoft.com/en-us/dotnet/api/system.array?view=net-6.0

I think String and Dictionary implement IEnumerable, but I guess as long as we check for string and Dictionary first it's safe.

My thought was to start small and expand this check as our graph test suite expands, thus the `TODO`.